src/CodeGen_Hexagon.cpp

```diff
@@ -421,6 +421,45 @@ Stmt inject_hvx_lock_unlock(Stmt body, const Target &target) {
     return body;
 }
 
+// Replace Div Nodes with long division if lanes > 1.
+class LongDiv : public IRMutator {
```

Comment:
Should this just be part of another existing mutator? Or maybe even just `CodeGen_Hexagon::visit(const Div*)`?

Reply:
I was initially thinking of making it a part of OptimizePatterns so as to take advantage of the peephole optimizations.